packages/api/tests/lib/test-granules.js

```diff
@@ -1,12 +1,203 @@
+    },
+  };
+  const granuleId = randomId('granule');
+  const files = [
+    {
+      bucket: s3Buckets.protected.name,
+      fileName: `${granuleId}.hdf`,
+      key: `${randomString(5)}/${granuleId}.hdf`,
+    },
+    {
+      bucket: s3Buckets.protected.name,
+      fileName: `${granuleId}.cmr.xml`,
+      key: `${randomString(5)}/${granuleId}.cmr.xml`,
+    },
+    {
+      bucket: s3Buckets.public.name,
+      fileName: `${granuleId}.jpg`,
+      key: `${randomString(5)}/${granuleId}.jpg`,
+    },
+  ];
+
+  const newGranule = fakeGranuleFactoryV2({ granuleId: granuleId, status: 'failed' });
+  newGranule.published = published;
+  newGranule.files = files;
+
+  await createS3Buckets([
+    s3Buckets.protected.name,
+    s3Buckets.public.name,
+  ]);
+
+  // Add files to S3
+  await Promise.all(newGranule.files.map((file) => s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: `test data ${randomString()}`,
+  })));
+
+  // create a new Dynamo granule
+  await granuleModel.create(newGranule);
+
+  // create a new PG granule
+  const newPGGranule = fakeGranuleRecordFactory(
+    {
+      granule_id: granuleId,
+      status: 'failed',
+      collection_cumulus_id: collectionCumulusId,
+    }
+  );
+  newPGGranule.published = published;
+  const [granuleCumulusId] = await granulePgModel.create(dbClient, newPGGranule);
+
+  // create PG files
+  await Promise.all(
+    files.map((f) => {
+      const pgFile = {
+        granule_cumulus_id: granuleCumulusId,
+        bucket: f.bucket,
+        file_name: f.fileName,
+        key: f.key,
+      };
+
+      return filePgModel.create(dbClient, pgFile);
+    })
+  );
+
+  return {
+    newPgGranule: await granulePgModel.get(dbClient, { cumulus_id: granuleCumulusId }),
+    newDynamoGranule: await granuleModel.get({ granuleId: newGranule.granuleId }),
+    files: files,
+  };
+}
+
+test.before(async (t) => {
+  process.env = {
+    ...process.env,
+    ...localStackConnectionEnv,
+    PG_DATABASE: testDbName,
+  };
+
+  // create a fake bucket
+  await createBucket(process.env.system_bucket);
+
+  // create fake Collections table
+  collectionModel = new models.Collection();
+  await collectionModel.createTable();
+
+  // create fake Granules table
+  granuleModel = new models.Granule();
+  await granuleModel.createTable();
+
+  granulePgModel = new GranulePgModel();
+  filePgModel = new FilePgModel();
+
+  const { knex, knexAdmin } = await generateLocalTestDb(testDbName, migrationDir);
+  t.context.knex = knex;
+  t.context.knexAdmin = knexAdmin;
+
+  // Create a Dynamo collection
+  // we need this because a granule has a fk referring to collections
+  t.context.testCollection = fakeCollectionFactory({
+    name: 'fakeCollection',
+    version: 'v1',
+    duplicateHandling: 'error',
+  });
+  await collectionModel.create(t.context.testCollection);
+
+  // Create a PG Collection
+  t.context.testPgCollection = fakeCollectionRecordFactory({ cumulus_id: collectionId });
```

Comment:
Same comment as elsewhere: I'd prefer not to arbitrary set the `cumulus_id` of this fixture directly and instead get the cumulusId when we create the object

```
const [collectionCumulusId] = await collectionPgModel.create(t.context.knex, t.context.testPgCollection);
t.context.collectionCumulusId = collectionCumulusId;
```

Reply:
👍 
db2bfb86046d4a280d8a6be84d27ae6449ec2cf5